### PR TITLE
Add marshaling functions for TPM structs

### LINKED ISSUE #192: Rewrite TPM marshaling functions as macros
There is a lot of redundant code around marshaling the various TPM structs. These cases could probably be covered by a macro or perhaps some more generic function or trait.

Relevant functions in `tpm.rs` include:
`tpm_digest_to_vec`
`tpm_sig_to_vec`
`tpm_att_to_vec`

There is also marshaling in `create_ek()` that could use the future function.

### LINKED ISSUE #194: PCR output file is incorrect
Even basing it on [this function](https://github.com/tpm2-software/tpm2-tools/blob/master/tools/tpm2_quote.c#L47-L91), two errors are returned:

`ERROR: Failed to read PCR selection from file` from this [line](https://github.com/tpm2-software/tpm2-tools/blob/f152cd0777ff6874389eef7ef94f2e95933e1f54/tools/misc/tpm2_checkquote.c#L260)
(from the identity quote handler)

`ERROR: Malformed PCR file, pcr count cannot be greater than 32, got: 4294967295` from this [line](https://github.com/tpm2-software/tpm2-tools/blob/f152cd0777ff6874389eef7ef94f2e95933e1f54/tools/misc/tpm2_checkquote.c#L271)
(from the integrity quote handler)

It's strange that these errors are different, as they are handled by the same code.

### PR DESCRIPTION
**Pull request by lkatalin**
Fixes #192 
Also fixes #194 with many thanks to @puiterwijk for debugging

**### Please start with reviewing this PR before the others!**

### CODE REVIEW

**Comment by lkatalin** [comment]
@lukehinds @puiterwijk @ashcrow Not sure if this will ping you since it's been open for a while, but switched from draft because it's ready for review whenever someone has a few moments. Thanks!

File: src/tpm.rs [hunk 1]
```diff
@@ -83,26 +87,85 @@ pub(crate) fn create_ek(
...
+// These marshaling functions use the macro above and are based on this format:
+// https://github.com/fedora-iot/clevis-pin-tpm2/blob/main/src/tpm_objects.rs#L64
+// ... and on these marshaling functions:
+// https://github.com/parallaxsecond/rust-tss-esapi/blob/main/tss-esapi-sys/src/ \
+// bindings/x86_64-unknown-linux-gnu.rs#L16010
+//
+// Functions can be created using the following form:
+// create_marshal_fn!(name_of_function_to_create, struct_to_be_marshaled, marshaling_function);
+//
+create_marshal_fn!(pub_to_vec, TPM2B_PUBLIC, Tss2_MU_TPM2B_PUBLIC_Marshal);
```
Comment (ashcrow) [inline src/tpm.rs:135]: I confuse myself with macros. Is `create_marshal_fn` creating `pub_to_vec` which is used later?

Comment (lkatalin) [reply src/tpm.rs:135]: @ashcrow Yeah, exactly - there are slightly different functions that all have the same basic structure. In all of the functions, some struct `TPM2B_something` needs to be turned into a vector and has a corresponding marshaling function. The macro can create all of these functions without replicating a bunch of code with only slight variations.

`pub_to_vec` and the other functions created here are used in the next PR on top of this one.

**Review by ashcrow — APPROVED** [review]
Seems reasonable to me!

**Pull request merged**